examples/client/object_server.c

```diff
@@ -224,6 +224,50 @@ static uint8_t prv_get_value(lwm2m_data_t * dataP,
     }
 }
 
+static void remove_optional_resource(uint16_t *resList, int val, int id, int *nbRes) {
+    int i;
+    if (val < 0) {
+        for (i = 0; i < *nbRes; i++) {
+            if (resList[i] == id) {
+                *nbRes -= 1;
+                memmove(&resList[i], &resList[i + 1], (*nbRes - i) * sizeof(resList[i]));
+                break;
+            }
+        }
+    }
+}
+
+static void setup_resources(lwm2m_data_t *const *dataArrayP, const uint16_t *resList, int nbRes) {
```

Comment:
```suggestion
static void setup_resources(lwm2m_data_t *const *dataArrayP, const uint16_t *const resList, const int nbRes) {
```

Reply:
Also, while on it, would change `int nbRes` to `size_t nbRes` and clean up the caller sites. While extra work, this seems quite little effort.

Feel free to ignore however, as this would also make a perfect follow up PR.